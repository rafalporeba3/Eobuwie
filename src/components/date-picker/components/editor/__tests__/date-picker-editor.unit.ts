import { createLocalVue, Wrapper }   from '@vue/test-utils';
import Vuex                          from 'vuex';
import CompositionApi                from '@vue/composition-api';
import { createTestInstanceShallow } from '@/__tests__/factory';
import cArrowRightIcon               from '@/components/icons/arrow-right.component.vue';
import cDatePickerEditor             from '../date-picker-editor.component.vue';

const localVue = createLocalVue();
localVue.use(CompositionApi);
localVue.use(Vuex);

describe('Date picker editor component', () => {
  let wrapper!: Wrapper<any>;

  beforeEach(() => {
    const { wrapperInstance } = createTestInstanceShallow(cDatePickerEditor, {
      localVue,
      propsData: {
        startPlaceholder: 'Start placeholder',
        endPlaceholder: 'End placeholder',
        isPanelVisible: false,
      },
    });

    wrapper = wrapperInstance;
  });

  it('should render cDatePickerEditor component with correct class', () => {
    expect(wrapper.classes())
      .toContain('c-date-picker-editor');
  });

  it('should add --active class to component when isPanelVisible is true', async () => {
    await wrapper.setProps({ isPanelVisible: true });

    expect(wrapper.classes())
      .toContain('c-date-picker-editor--active');
  });

  it('should display correct placeholders', () => {
    expect(wrapper.props('startPlaceholder')).toBe('Start placeholder');
    expect(wrapper.props('endPlaceholder')).toBe('End placeholder');
  });

  it('should display default placeholders when placeholder props are not defined', async () => {
    await wrapper.setProps({ startPlaceholder: undefined, endPlaceholder: undefined });
    expect(wrapper.props('startPlaceholder')).toBe('Start');
    expect(wrapper.props('endPlaceholder')).toBe('End');
  });

  it('should contain cArrowRightIcon', () => {
    expect(wrapper.findComponent(cArrowRightIcon)
      .exists())
      .toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
