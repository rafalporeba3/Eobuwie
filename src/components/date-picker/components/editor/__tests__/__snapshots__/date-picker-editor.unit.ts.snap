// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date picker editor component should match snapshot 1`] = `
<div class="c-date-picker-editor"><span class="c-date-picker-editor__placeholder">Start placeholder</span>
  <c-arrow-right-icon-stub></c-arrow-right-icon-stub> <span class="c-date-picker-editor__placeholder">End placeholder</span>
</div>
`;
